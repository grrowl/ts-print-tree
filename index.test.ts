import { jest, describe, expect, test } from "@jest/globals";
import { tree } from ".";

describe("tree module", () => {
  test("calls tree() and matches console.log snapshot", () => {
    console.log = jest.fn();

    tree();
    expect((console.log as jest.Mock).mock.calls).toMatchSnapshot();
  });
});
