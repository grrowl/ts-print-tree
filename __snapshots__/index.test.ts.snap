// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tree module calls tree() and matches console.log snapshot 1`] = `
[
  [
    "├─ __snapshots__/
├─ dist/
├─ src/
│  ├─ cli.ts
│  └─ index.ts
      └─ readTsConfig(rootDir: string): ParsedCommandLine
      └─ createProgram(rootDir: string): Program
      └─ analyzeFile(sourceFile: SourceFile, typeChecker: TypeChecker): string
      └─ traverseDirectory(dir: string, program: Program, shouldIgnore: (path: string) => boolean, prefix: string): string
      └─ shouldIgnore(ignoredPatterns: (string | RegExp)[], pathToCheck: string): boolean
      └─ tree(rootDir: string, ignored: (string | RegExp)[]): void
",
  ],
]
`;
